src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -7544,6 +7544,20 @@ we will have to use something else.
                     }
                 }
             }
+        } else if (AST::is_a<AST::List_t>(*at->m_value)) {
+            AST::List_t* clist = AST::down_cast<AST::List_t>(at->m_value);
+            visit_List(*clist);
+            if (!tmp) {
+                throw SemanticError("cannot call " + std::string(at->m_attr) + " on an empty list" , loc);
+            }
```

Comment:
@Shaikh-Ubaid what is a better way to handle empty lists at compile-time?

Reply:
I think the current approach is fine.